sdk/nodejs/appservice/zMixins.ts

```diff
@@ -394,7 +394,7 @@ async function produceDeploymentArchiveAsync(args: MultiCallbackFunctionAppArgs)
         tracing: { consoleLevel: "verbose" },
         extensionBundle: {
             id: "Microsoft.Azure.Functions.ExtensionBundle",
-            version: "[1.*, 2.0.0)"
+            version: "[3.3.0, 4.0.0)"
```

Comment:
As you pointed out on #1201, there's an enforced mapping between extension bundle and runtime version. If we change this, won't we break users that are currently using pulumi-azure with 1.x functions?

Reply:
@thomas11,
How about now? Just a simple map that should get it going for all versions?

Reply:
Thanks Thomas,

Please find answers to your points below:

* "I see the previous default version was ~3 but the bundle version was [1...]. How did that work if the mapping is strict, shouldn't ~3 have been matched with [3.3....]? Or did the default path just never work?"
> I remember reading somewhere from the docs, that for the previous versions of Function Apps the extension bundle version was not required to be specified. However, this was changed whit the release of v4. 

* "Can you please put a comment at the mapping table explaining why we need this, and maybe a link to the MS docs?"
> Sure, I can do that.

* "How can we test this? Did you test it on your end with a ~4 function?"
> I have tested this with v3 and v4 functions - working as expected.

* "I wonder if we should err on the safe side and default to ~1 (or the previous ~3), to avoid breaking existing users."
> I am not sure if this is going to break something. If users are following best practices they should check the preview carefully before actually applying any change. I can confirm that the diff in the preview is clear enough to sign that the version is being changed. They can either act accordingly and set version ~3 or consider migrating to ~4. Especially when v3 is EOL now and no service support is being provided.
The current default version value was set to ~3, so if there are users out there that are using v1 - then they have probably already specified ``version: "~1"``

Reply:
@thomas11, Is there anything else we need to discuss?

Reply:
I think we're all good! Thank you so much for the contribution!

Reply:
Thank you @thomas11